server/container_create.go

```diff
@@ -798,19 +807,52 @@ func (s *Server) createSandboxContainer(ctx context.Context, containerID string,
 	specgen.HostSpecific = true
 	specgen.ClearProcessRlimits()
 
-	var readOnlyRootfs bool
+	readOnlyRootfs := s.config.ReadOnly
+
 	var privileged bool
 	if containerConfig.GetLinux().GetSecurityContext() != nil {
 		if containerConfig.GetLinux().GetSecurityContext().Privileged {
 			privileged = true
 		}
-
 		if containerConfig.GetLinux().GetSecurityContext().ReadonlyRootfs {
 			readOnlyRootfs = true
-			specgen.SetRootReadonly(true)
 		}
 	}
+	specgen.SetRootReadonly(readOnlyRootfs)
 
+	if s.config.ReadOnly {
+		options := []string{"rw", "noexec", "nosuid", "nodev", "tmpcopyup"}
+		if !isInCRIMounts("/run", containerConfig.GetMounts()) {
+			mnt := rspec.Mount{
+				Destination: "/run",
+				Type:        "tmpfs",
+				Source:      "tmpfs",
+				Options:     append(options, "mode=0755", "size=65536k"),
```

Comment:
I would much rather not hard-code this size.  Is there no way to get this from the CRI mount request?

Reply:
Now that I check /run on the host has no size specified.  So we should just remove it.

Reply:
> A configurable tmpfs size for each tmpfs we create.

We already have [`DefaultShmSize`][1] (initially from #249) in this direction.  Maybe continue down that path for these other tmpfs?

[1]: https://github.com/kubernetes-incubator/cri-o/blob/b0f6d98b9435c094a24fe55c3e75f3daff7196ee/lib/sandbox/sandbox.go#L172-L173

Reply:
Removed the size.